qualtran/bloqs/data_loading/one_hot_encoding.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import attrs
+import cirq
+from numpy._typing import NDArray
+
+from qualtran import GateWithRegisters, QAny, QUInt, Signature
+from qualtran.bloqs.basic_gates import TwoBitCSwap
+
+
+@attrs.frozen
+class OneHotEncoding(GateWithRegisters):
+    """
+    One-hot encode a binary integer into a target register.
+
+    Registers:
+        a: an unsigned integer
+        b: the target to one-hot encode.
+
+    References:
+        [Windowed quantum arithmetic](https://arxiv.org/pdf/1905.07682.pdf)
+        Figure 4
+    """
+
+    binary_bitsize: int
```

Comment:
Can we just call this `bitsize`? that's what we do elsewhere. the "bi" in "bitsize" means "bi"nary

Reply:
I guess this is to contrast with the unary bitsize. Maybe it's worth disambiguating. I defer to your judgement. But please document this attribute in an `Args:` section in the class docstring